source/solvers/mf_navier_stokes.cc

```diff
@@ -1192,12 +1199,60 @@ MFNavierStokesPreconditionGMG<dim>::initialize(
              .preconditioner ==
            Parameters::LinearSolver::PreconditionerType::gcmg)
     {
+      if (this->minlevel != this->intlevel)
+        {
+          // Create main MG object
+          this->mg_intermediate =
+            std::make_shared<Multigrid<VectorType>>(*this->mg_matrix,
+                                                    *this->mg_coarse,
+                                                    *this->mg_transfer_gc,
+                                                    *this->mg_smoother,
+                                                    *this->mg_smoother,
+                                                    this->minlevel,
+                                                    this->intlevel);
+
+          // Create MG preconditioner
+          this->gc_multigrid_preconditioner_intermediate =
+            std::make_shared<PreconditionMG<dim, VectorType, GCTransferType>>(
+              this->dof_handler, *this->mg_intermediate, *this->mg_transfer_gc);
+
+          if (false)
+            this->coarse_grid_solver_control_intermediate =
+              std::make_shared<IterationNumberControl>(100,
+                                                       1e-20,
+                                                       false,
+                                                       false);
+          else
+            this->coarse_grid_solver_control_intermediate =
+              std::make_shared<ReductionControl>(
+                1000, 1e-20, 1e-2, false, false);
```

Comment:
Do you want to keep this while we are doing temporary test or does this have a peculiar function? Because if(false) clearly won,t happen ^^!